tests/lib/rules/no-whitespace-before-property.js

```diff
@@ -507,6 +510,46 @@ ruleTester.run("no-whitespace-before-property", rule, {
             code: "foo ['bar ' + 1 + ' baz']",
             output: "foo['bar ' + 1 + ' baz']",
             errors: ["Unexpected whitespace before property 'bar ' + 1 + ' baz'."]
+        },
+        {
+            code: "5       .toExponential()",
+            output: "5 .toExponential()",
```

Comment:
Another option would be to consider `5 .toExponential()` invalid, and fix it to `5..toExponential()`. I argued against this change [here](/eslint/eslint/pull/7185#issuecomment-247813055), but I think I changed my mind about this; the rule is called `no-whitespace-before-property`, and `5 .toString()` has whitespace before the property, so it seems more intuitive for it to be disallowed.